### ISSUE TITLE
"Short read on header, read 0, 4 bytes expected."

### ISSUE THREAD

**Issue by RickvdP** [php-riak/riak-client#17]
Hi @FabioBatSilva,

Another issue I came across. I'm working multiple workers in parallel pushing data into Riak, and from time to time see a lot (or all) workers crashing with the error below. I am not sure if there is anything to be done about this, or that the Riak cluster just gets overwhelmed somehow?

```
[2016-04-24 15:25:55] consumer76.ERROR: Something went wrong when storing object into Riak: exception 'Riak\Client\Core\Transport\RiakTransportException' with message 'Short read on header, read 0, 4 bytes expected.' in /root/riak-migration/vendor/php-riak/riak-client/src/Riak/Client/Core/Transport/Proto/ProtoConnection.php:151 
Stack trace: 
#0 /root/riak-migration/vendor/php-riak/riak-client/src/Riak/Client/Core/Transport/Proto/ProtoConnection.php(117): Riak\Client\Core\Transport\Proto\ProtoConnection->receiveLengthHeader(Object(Riak\Client\Core\Transport\Proto\ProtoStream)) 
#1 /root/riak-migration/vendor/php-riak/riak-client/src/Riak/Client/Core/Transport/Proto/ProtoClient.php(96): Riak\Client\Core\Transport\Proto\ProtoConnection->receive(Object(Riak\Client\Core\Transport\Proto\ProtoStream)) 
#2 /root/riak-migration/vendor/php-riak/riak-client/src/Riak/Client/Core/Transport/Proto/ProtoClient.php(67): Riak\Client\Core\Transport\Proto\ProtoClient->receiveMessage(Object(Riak\Client\Core\Transport\Proto\ProtoStream), 12) 
#3 /root/riak-migration/vendor/php-riak/riak-client/src/Riak/Client/Core/Transport/Proto/Kv/ProtoPut.php(113): Riak\Client\Core\Transport\Proto\ProtoClient->send(Object(Riak\Client\ProtoBuf\RpbPutReq), 11, 12) 
#4 /root/riak-migration/vendor/php-riak/riak-client/src/Riak/Client/Core/RiakProtoTransport.php(87): Riak\Client\Core\Transport\Proto\Kv\ProtoPut->send(Object(Riak\Client\Core\Message\Kv\PutRequest)) 
#5 /root/riak-migration/vendor/php-riak/riak-client/src/Riak/Client/Core/Operation/Kv/StoreOperation.php(61): Riak\Client\Core\RiakProtoTransport->send(Object(Riak\Client\Core\Message\Kv\PutRequest)) 
#6 /root/riak-migration/vendor/php-riak/riak-client/src/Riak/Client/Core/RiakNode.php(41): Riak\Client\Core\Operation\Kv\StoreOperation->execute(Object(Riak\Client\Core\RiakProtoTransport)) 
#7 /root/riak-migration/vendor/php-riak/riak-client/src/Riak/Client/Core/RiakCluster.php(109): Riak\Client\Core\RiakNode->execute(Object(Riak\Client\Core\Operation\Kv\StoreOperation)) 
#8 /root/riak-migration/vendor/php-riak/riak-client/src/Riak/Client/Command/Kv/StoreValue.php(81): Riak\Client\Core\RiakCluster->execute(Object(Riak\Client\Core\Operation\Kv\StoreOperation)) 
#9 /root/riak-migration/vendor/php-riak/riak-client/src/Riak/Client/RiakClient.php(61): Riak\Client\Command\Kv\StoreValue->execute(Object(Riak\Client\Core\RiakCluster)) 
#10 /root/riak-migration/worker.php(38): Riak\Client\RiakClient->execute(Object(Riak\Client\Command\Kv\StoreValue)) 
#11 /root/riak-migration/worker.php(149): storeInRiak('Bucket', 'Z2ssssss...', 'application/oct...', 'x\x9C\xBDU\xFBS\xDBF\x10\x96A\x16\x06\xE3#...') 
#12 [internal function]: process_message(Object(PhpAmqpLib\Message\AMQPMessage)) #13 /root/riak-migration/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AMQPChannel.php(983): call_user_func('process_message', Object(PhpAmqpLib\Message\AMQPMessage)) 
#14 [internal function]: PhpAmqpLib\Channel\AMQPChannel->basic_deliver(Object(PhpAmqpLib\Wire\AMQPReader), Object(PhpAmqpLib\Message\AMQPMessage)) 
#15 /root/riak-migration/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(202): call_user_func(Array, Object(PhpAmqpLib\Wire\AMQPReader), Object(PhpAmqpLib\Message\AMQPMessage)) 
#16 /root/riak-migration/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(341): PhpAmqpLib\Channel\AbstractChannel->dispatch('60,60', '\nconsumer76\x00\x00\x00\x00...', Object(PhpAmqpLib\Message\AMQPMessage)) 
#17 /root/riak-migration/worker.php(186): PhpAmqpLib\Channel\AbstractChannel->wait() 
#18 {main}
```

**Comment by FabioBatSilva** [comment]
@RickvdP Do you see anything on `/path-riak/log/{error,crash}.log` when this happen ?
